packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -217,10 +219,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
   React.useEffect(() => {
     setContainerWidth(rootRef.current!.clientWidth);
```

Comment:
On React 17 we did not have automatic batching, so maybe it's  better to have a shared state for the width and the height